src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
 'use strict';
 
-// write your code here
+const container = document.querySelector('.container');
+const controls = container.querySelector('.controls');
+const startButton = controls.querySelector('.start');
+const gameField = container.querySelector('.game-field');
+const messages = container.querySelector('.message-container');
+const startMessage = messages.querySelector('.message-start');
+// const loseMessage = messages.querySelector('.message-lose');
+const winMessage = messages.querySelector('.message-win');
+const scoreValue = container.querySelector('.game-score');
+let arrField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+// Create restart button
+
+const restartButton = document.createElement('button');
+
+function createRestartButton() {
+  restartButton.classList.add('restart', 'button');
+  restartButton.textContent = 'Restart';
+
+  return restartButton;
+}
+
+// Create two random item
+
+function createTwoNewItem() {
+  newRandomCell();
+  newRandomCell();
+}
+
+// Change start button to restart
+
+startButton.addEventListener('click', event => {
+  startButton.remove();
+  controls.append(createRestartButton());
+  startMessage.classList.add('hidden');
+  createTwoNewItem();
+  calculateScore();
+  loopArray(buildGameField);
+});
+
+// Reset game field
+
+restartButton.addEventListener('click', event => {
+  arrField = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+  createTwoNewItem();
+  calculateScore();
+  loopArray(buildGameField);
+});
+
+// Win message
+
+function checkWin(number) {
+  if (number === 2048) {
+    winMessage.classList.remove('hidden');
+  }
+}
+
+// Build game field
+
+function buildGameField(row, column) {
+  const fieldCell = gameField.rows[row].cells[column];
+
+  if (arrField[row][column] > 0) {
+    fieldCell.classList = ['field-cell'];
+    fieldCell.classList.add(`field-cell--${arrField[row][column]}`);
+    fieldCell.textContent = `${arrField[row][column]}`;
+  } else {
+    fieldCell.classList = ['field-cell'];
+    fieldCell.textContent = '';
+  }
+}
+
+// 10% random
+
+function randomValue() {
+  if (Math.random() < 0.1) {
+    return 4;
+  } else {
+    return 2;
+  }
+}
+
+// Random position
+
+function randomPosition() {
+  return {
+    row: Math.floor(Math.random() * 4),
+    column: Math.floor(Math.random() * 4),
+  };
+}
+
+// Calculate score
+
+function calculateScore() {
+  let totalScore = 0;
+
+  for (const row of arrField) {
+    for (const column of row) {
+      totalScore += column;
+    }
+  }
+
+  scoreValue.textContent = totalScore;
+}
+
+// New random cell
+
+function newRandomCell() {
+  const { row, column } = randomPosition();
+
+  if (arrField[row][column] === 0) {
```

Comment:
таке теж виносять в змінну

```suggestion
  const isCellEmpty =  arrField[row][column] === 0;

  if (isCellEmpty) {
```